spec/controllers/announcements_controller_spec.rb

```diff
@@ -15,6 +15,23 @@
     @app_config = FactoryGirl.create(:app_config)
   end
 
+  shared_examples 'can hide announcement' do
+    before do
+      @announcement = FactoryGirl.create(:announcement)
+      @attributes = FactoryGirl.attributes_for(:announcement)
+      request.env['HTTP_REFERER'] = 'where_i_came_from'
+      get :hide, id: @announcement
+    end
+    it 'should set some cookie values' do
+      verifier = ActiveSupport::MessageVerifier.new(
+        Reservations::Application.config.secret_key_base)
+      name = 'hidden_announcement_ids'
+      @request.cookies[name] = verifier.generate([@announcement[:id].to_s])
```

Comment:
Ok, the major issue is here. You're basically setting the relevant cookie in the *request* and therefore it's still set in the response, regardless of the controller behavior. If we look at this [SO answer](http://stackoverflow.com/a/5482517/2187922), we see how we would test for a signed cookie, effectively deleting line 29 (this line) and adding the following after line 30:

```
jar.signed[name] = [@announcement[:id].to_s]
```

This gives us an equivalent cookie to test against, and then we check to make sure the response has the right cookie value. I made sure it worked by commenting out the relevant line in `ability.rb` and the test failed as expected.

Let's get this text fixed and duplicate it to check all user roles. Let me know if you have any questions!